defaults/main.yml

```diff
@@ -24,8 +24,8 @@ postfix_mynetworks: "127.0.0.0/8"
 # postfix_relay_domains: "$mydestination"
 
 # If you want to forward emails to another central relay server, set relayhost.
-# use brackets to sent to the A-record of the relayhost.
-# postfix_relayhost: [relay.example.com]
+# brackets are automatically added to sent to the A-record of the relayhost, IP or dns name.
+# postfix_relayhost: relay.example.com
```

Comment:
I would like to keep the default postfix behaviour; with brackets indicates a host, without, lookup the mx-records of a domain.

Reply:
ok try the following:
make a var called relay_host  = my_server.domain.lan in your inventory vars
then set that var on your role like postfix_relayhost: [{{ relay_host }}]
you will see that the output of your template file will be in your main.cf
relay_host = ['relay.example.com']
this is due to how python interprets the [{ combo as it thinks it's an array

I moved the [] down to the jinja template to work around that bug. as jinja doesn't care.
everyone uses or ip or a DNS name for a relay host, I've yet to see someone use UUCP.

if you know a better way to work around it?

Reply:
ok i got confirmation from the ansible team
if you do [{{var}}] ansible will think its an array and pipe it to jinja as multiple values 'var'
so if you absolutely don't want the [] moved to the jinja file, the only thing I can think about is a jinja regex remove of the '' as no domain or ip should ever have them?
what do you think?

Reply:
But, I'd like to keep the behavior of Postfix similar to what is described in [the manual of postfix on relayhosts](http://www.postfix.org/BASIC_CONFIGURATION_README.html#relayhost).

```
# This sends the mail to the mx-records of example.com:
relay_host: 'example.com'

# This sends the mail to the a-records of example.com:
relay_host: '[example.com]'
```

It's a functional difference that I'd like to keep.

Reply:
ok, the latest commit reverts back the [] to what you had. And I've added a filter to the jinja template to search for the single quote bug and remove the characters.
This way you can have both examples of what you have and still use the variable replacement